tests/consensus_spec/test_fixture_kzg.nim

```
@@ -305,6 +308,171 @@ proc runRecoverCellsAndKzgProofsTest(suiteName, suitePath, path: string) =
+        continue
+
+      var
+        shouldSkip = false
+        invalidData: seq[MatrixEntry]
+        indices: seq[ColumnIndex]
+
+      for i in 0..<inputCells.len:
+        let
+          cellBytes = fromHex[2048](inputCells[i].getStr).valueOr:
+            # skip case of cell has length != 2048
+            # TODO: support dynamic length for future-proof
+            shouldSkip = true
+            break
+          columnIdx = ColumnIndex(inputIndices[i].getInt.toUInt64.get)
+
+        # Duplicate index is not compatible with `seq[DataColumnSidecar]`
+        if columnIdx in indices:
+          shouldSkip = true
+          break
+
+        indices.add(columnIdx)
+        invalidData.add(MatrixEntry(
+          cell: Cell(bytes: cellBytes),
+          column_index: columnIdx,
+          row_index: RowIndex(validCount)))
+
+      if shouldSkip:
+        continue
+
+      test "KZG - Recover Cells And Kzg Proofs Parallel - invalid - " & path.rsplit("_case_invalid_", 1)[1]:
+        let
+          input = validData.filterIt(it.column_index in indices) & invalidData
+          rowCount = validCount + 1
+
+        # convert input into column
+        var
+          colInput: seq[ref DataColumnSidecar]
+          colEntries: Table[uint64, seq[MatrixEntry]]
+
+        for entry in input:
+          colEntries.mgetOrPut(entry.column_index, @[]).add(entry)
+
+        for cIdx, entries in colEntries:
+          var cells = newSeq[Cell](rowCount)
+          for entry in entries:
+            cells[entry.row_index] = entry.cell
+          let sidecar = DataColumnSidecar(
+            index: ColumnIndex(cIdx),
+            column: DataColumn(cells))
+          colInput.add(newClone(sidecar))
+
+        # check recovered cells and proofs
+        var tp = Taskpool.new()
+        let v = tp.recover_cells_and_proofs_parallel(colInput)
+        check v.isErr
+  except Exception:
+    debugEcho "Problem in loading KZG invalid case data"
```

Don't need `debugEcho`, since it's a `proc`. Can use `echo`.

But that's not even really the right approach. this is a test runner, not application code. If there's a problem loading the KZG invalid case data, the test has failed. Maybe it's failed because the test runner is buggy, maybe it's failed because the test data isn't available, maybe it's failed because the test data generation was botched. It doesn't really matter, swallowing the an exception [1] isn't the correct way of dealing with this.

Furthermore, `Exception` isn't the best error to catch here. If one wants as broad an umbrella as possible, `CatchableError`, but really, tests are a place it's ok for things to fail, that's the point, and any kind of broad exception handling hides/hinders that.

---

Sure will address this.

Regards to the test runner, it seems that try-catch is needed or it wouldn't compile when the function is out of `test:` template call. Would be great to know if there is any workaround as I prefer not to use try-catch in test as well.

---

- https://github.com/status-im/nimbus-eth1/pull/3658

seems to have addressed this in the `column-syncer` branch.